axum-debug/src/lib.rs

```diff
@@ -251,6 +251,24 @@ mod debug_handler {
         };
         let span = ty.span();
 
+        let declare_inputs = item_fn
+            .sig
+            .inputs
+            .iter()
+            .filter_map(|arg| match arg {
+                FnArg::Receiver(_) => None,
+                FnArg::Typed(pat_ty) => {
+                    let pat = &pat_ty.pat;
+                    let ty = &pat_ty.ty;
+                    Some(quote! {
+                        let #pat: #ty = panic!();
```

Comment:
I briefly thought that this could cause the same issue in a way but I don't think it can, since handler functions can't practically be generic (right?)

Reply:
Handlers can be generic but `debug_handler` doesn’t support that and panics. There might be ways to make it work but it’s a huge can of worms.

Reply:
Oh I guess it works with an explicit generic parameter and registering it under a route as `handler::<Foo, Bar>`. I was thinking of `impl Trait` in argument position.